### PR TITLE
Add step to trigger gitops to update image

### PR DESCRIPTION
**Pull request by michel-laterman**
Trigger update fleet-server image update through gitops.

### CODE REVIEW

File: .buildkite/pipeline.yml [hunk 1]
```diff
@@ -98,6 +98,21 @@ steps:
       - step: "tests"
         allow_failure: false
 
+  - label: ":argo: Update image tag for Fleet Server"
+    trigger: serverless-gitops-update-stack-image-tag
+    async: true
+    branches: main
+    build:
+      env:
+        IMAGE_TAG: "$DOCKER_IMAGE_TAG"
```
Comment (michel-laterman) [inline .buildkite/pipeline.yml:107]: I'm not sure if this is actually the right tag

File: .buildkite/pipeline.yml [hunk 2]
```diff
@@ -98,6 +98,21 @@ steps:
...
+        SERVICE: fleet-controller
+        NAMESPACE: fleet-ci
+        IMAGE_NAME: fleet-server
+        COMMIT_MESSAGE: "gitops: update fleet-server tag to elastic/fleet-server@$GIT_ABBREV_COMMIT"
```
Comment (mrodm) [inline .buildkite/pipeline.yml:111]: I think `GIT_ABBREV_COMMIT` env. variable does not exist in the Buildkite context.

Probably, it should be used:
- GITHUB_PR_HEAD_SHA
- BUILDKITE_COMMIT

They contain the full sha of the commit. Example:
```
GITHUB_PR_HEAD_SHA="74556f61420168ef34592455a2bd42e4c4e01dc9"
BUILDKITE_COMMIT="74556f61420168ef34592455a2bd42e4c4e01dc9"
```

File: .buildkite/pipeline.yml (hunk 1 above)
Comment (mrodm) [reply .buildkite/pipeline.yml:107]: This environment variable does not exist in Buildkite or when the execution reaches this step.

What value should it be here ?

File: .buildkite/pipeline.yml (hunk 2 above)
Comment (mrodm) [reply .buildkite/pipeline.yml:111]: If those values needs to be calculated, it looks like it should be created this step dynamically (following the example in Kibana repo).

Here there is another example of creating steps dynamically:
- https://github.com/elastic/elastic-package/blob/main/.buildkite/pipeline.yml#L38-L39
- https://github.com/elastic/elastic-package/blob/main/.buildkite/pipeline.trigger.integration.tests.sh

File: .buildkite/pipeline.yml [hunk 3]
```diff
@@ -98,6 +98,21 @@ steps:
...
 
+  - label: ":argo: Update image tag for Fleet Server"
+    trigger: serverless-gitops-update-stack-image-tag
+    async: true
+    branches: main
+    build:
+      env:
+        IMAGE_TAG: "$DOCKER_IMAGE_TAG"
+        SERVICE: fleet-controller
+        NAMESPACE: fleet-ci
```
Comment (mrodm) [inline .buildkite/pipeline.yml:109]: What does this namespace refer to ? Is it to kubernetes namespace or the docker namespace (URL) ? 

Should it be `observability-ci` ? 
```suggestion
        NAMESPACE: observability-ci
```

This is where docker images are being pushed right now by Jenkins: https://github.com/elastic/fleet-server/blob/main/.ci/Jenkinsfile#L19

**Review by mrodm — APPROVED** [review]
LGTM

**Review by jlind23 — APPROVED** [review]

**Pull request merged**